scripts/pi-hole/php/savesettings.php

```diff
@@ -233,6 +233,26 @@ function readAdlists()
 					$extra .= "no-dnssec";
 				}
 
+				// Check if Conditional Forwarding is requested
+				if(isset($_POST["conditionalForwarding"]))
+				{
+					// Validate conditional forwarding IP
+					if (!validIP($_POST["conditionalForwardingIP"]))
+					{
+						$error .= "Conditional forwarding IP (".htmlspecialchars($_POST["conditionalForwardingIP"]).") is invalid!<br>";
+					}
+
+					// Validate conditional forwarding domain name
+					if(!validDomain($_POST["conditionalForwardingDomain"]))
+					{
+						$error .= "Conditional forwarding domain name (".htmlspecialchars($_POST["conditionalForwardingDomain"]).") is invalid!<br>";
+					}
+
+					$addressArray = explode(".", $_POST["conditionalForwardingIP"]);
+					$reverseAddress = $addressArray[2].".".$addressArray[1].".".$addressArray[0].".in-addr.arpa";
+					$extra .= " conditional_forwarding ".$_POST["conditionalForwardingIP"]." ".$_POST["conditionalForwardingDomain"]." $reverseAddress";
```

Comment:
You should put these lines under an if statement to make sure that the IP is valid so that the array accesses don't crash:
```php
if(!$error) {
    // ...
}
```